src/pages/todoSetup/hooks/useTodoSetup/useTodoSetup.js

```diff
@@ -10,29 +10,29 @@ import { createTodoItem } from './useTodoSetup.helpers'
 function useTodoSetup(todoList, updateTodoList, saveTodoList) {
     const [todoText, setTodoText] = useState("");
 
-       const handleTodoTextChange = (e) => {
+       const handleTodoTextChange = useCallback((e) => {
             setTodoText(e.target.value);
-        }
+        },[]);
 
-        const handleTodoItemSave = (e) => {
+        const handleTodoItemSave = useCallback((e) => {
             e.preventDefault();
             if (!todoText) return;
              
             const newTodoInfo = createTodoItem(todoList, todoText);
             updateTodoList(newTodoInfo);
             setTodoText("");
-        }
+        },[todoList, todoText]);
```

Comment:
Check once if this can result in re-rendering of form, because handleTodoItemSave is used as a prop for todoForm and it will change because  todoList is changed once you mark it as completed which can lead to rerender of form. That should not be the case because both are independent if you see.